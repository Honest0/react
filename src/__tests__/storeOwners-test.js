// @flow

const { printOwnersList } = require('./storeSerializer');

describe('Store owners list', () => {
  let React;
  let ReactDOM;
  let TestUtils;
  let store;

  const act = (callback: Function) => {
    TestUtils.act(() => {
      callback();
    });
    jest.runAllTimers(); // Flush Bridge operations
  };

  beforeEach(() => {
    store = global.store;
    store.collapseNodesByDefault = false;

    React = require('react');
    ReactDOM = require('react-dom');
    TestUtils = require('react-dom/test-utils');
  });

  it('should drill through intermediate components', () => {
    const Root = () => (
      <Intermediate>
        <Leaf key="children" />
      </Intermediate>
    );
    const Wrapper = ({ children }) => children;
    const Leaf = () => <div>Leaf</div>;
    const Intermediate = ({ children }) => <Wrapper>{children}</Wrapper>;

    act(() => ReactDOM.render(<Root />, document.createElement('div')));
    expect(store).toMatchSnapshot('1: mount');

    const rootID = store.getElementIDAtIndex(0);
    expect(
      printOwnersList(store.getOwnersListForElement(rootID))
    ).toMatchSnapshot('2: components owned by <Root>');

    const intermediateID = store.getElementIDAtIndex(1);
    expect(
      printOwnersList(store.getOwnersListForElement(intermediateID))
    ).toMatchSnapshot('3: components owned by <Intermediate>');
  });
});
