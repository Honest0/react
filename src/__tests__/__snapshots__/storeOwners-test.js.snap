// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store owners list should drill through intermediate components: 1: mount 1`] = `
[root]
  ▾ <Root>
    ▾ <Intermediate>
      ▾ <Wrapper>
          <Leaf key="children">
`;

exports[`Store owners list should drill through intermediate components: 2: components owned by <Root> 1`] = `
"  ▾ <Root>
    ▾ <Intermediate>
        <Leaf key=\\"children\\">"
`;

exports[`Store owners list should drill through intermediate components: 3: components owned by <Intermediate> 1`] = `
"  ▾ <Intermediate>
    ▾ <Wrapper>"
`;
